SW_Output_get_functions.c

```diff
@@ -2442,3 +2442,76 @@ void get_soiltemp_agg(OutPeriod pd)
 	}
 }
 #endif
+
+//------ eSW_Frozen
+#ifdef SW_OUTTEXT
+
+/**
+@brief Gets state (frozen/unfrozen) for each layer for when dealing with OUTTEXT.
+
+@param pd Period.
+*/
+void get_frozen_text(OutPeriod pd)
+{
+    LyrIndex i;
+    SW_SOILWAT_OUTPUTS *vo = SW_Soilwat.p_oagg[pd];
+
+    char str[OUTSTRLEN];
+    sw_outstr[0] = '\0';
+
+    ForEachSoilLayer(i)
+    {
+        sprintf(str, "%c%.*f", _Sep, OUT_DIGITS, vo->lyrFrozen[i]);
+        strcat(sw_outstr, str);
+    }
+}
+#endif
+
+#if defined(RSOILWAT)
+
+/**
+@brief Gets soil state (frozen/unfrozen) for when dealing with RSOILWAT.
+
+@param pd Period.
+*/
+void get_frozen_mem(OutPeriod pd)
+{
+    LyrIndex i;
+    SW_SOILWAT_OUTPUTS *vo = SW_Soilwat.p_oagg[pd];
+
+    RealD *p = p_OUT[eSW_Frozen][pd];
+    get_outvalleader(p, pd);
+
+    ForEachSoilLayer(i)
+    {
+        p[iOUT(i, pd)] = stValues.lyrFrozen[i];

```

Comment:
This needs to be the prepared output `vo->lyrFrozen[i]` (aggregated if output at a longer time step than daily) -- and not the daily valued `stValues.lyrFrozen[i]` from the simulations. `get_frozen_text()` uses the correct variable for output.